test/uk/gov/hmrc/channelpreferences/controllers/PreferenceControllerSpec.scala

```diff
@@ -84,6 +96,40 @@ class PreferenceControllerSpec extends PlaySpec with ScalaFutures with MockitoSu
       status(response) mustBe CONFLICT
     }
   }
+  "Calling Agent Enrolment Stub endpoint " should {
+    """return OK for any AgentReferenceNumber(ARN) and itsaId""" in new TestSetup {
+      when(
+        mockAuthConnector.authorise[Option[AffinityGroup]](any[Predicate](), any[Retrieval[Option[AffinityGroup]]]())(
+          any[HeaderCarrier](),
+          any[ExecutionContext]()))
+        .thenReturn(Future.successful(Some(AffinityGroup.Agent)))
+
+      val postData: JsValue = Json.parse(s"""
+                                            |{
+                                            |  "arn": "testARN",
+                                            |  "itsaId": "testItsaId"
+                                            |}
+      """.stripMargin)
+
+      val fakePostRequest = FakeRequest("POST", "", Headers("Content-Type" -> "application/json"), postData)
+      val response = controller.agentEnrolment().apply(fakePostRequest)
+      status(response) mustBe OK
+    }
+
+    """return 400 for missing AgentReferenceNumber(ARN) and itsaId""" in new TestSetup {
+      when(
+        mockAuthConnector.authorise[Option[AffinityGroup]](any[Predicate](), any[Retrieval[Option[AffinityGroup]]]())(
+          any[HeaderCarrier](),
+          any[ExecutionContext]()))
+        .thenReturn(Future.successful(Some(AffinityGroup.Agent)))
+
+      val postData: JsValue = JsString("")
```

Comment:
is it better if we have empty Json instead of JsString ?

Reply:
We can also have a test for unauthorised ?

Reply:
> is it better if we pass some Json or empty Json instead of JsString ?

Done..